googlecloud-plugins/src/main/java/co/cask/hydrator/plugin/source/BigQuerySource.java

```diff
@@ -0,0 +1,222 @@
+                                                    " example: SELECT TOP(corpus, 10) as title, COUNT(*) as " +
+                                                    "unique_words FROM [publicdata:samples.shakespeare], where " +
+                                                    "publicdata is the project name, smalples is the dataset name, " +
+                                                    "shakespare is the table name. This is optional, if empty, just " +
+                                                    "read the inputTable configured";
+    private static final String JSON_KEYFILE_DESC = "the credential json key file path";
+    private static final String INPUT_TABLE_DESC = "The BigQuery table to read from, in the form " +
+                                                   "[optional projectId]:[datasetId].[tableId]. Example: " +
+                                                   "publicdata:samples.shakespeare. Note that if the import query is " +
+                                                   "specified, this table should be a empty table with the " +
+                                                   "query result schema. User need to first create such a table";
+    private static final String OUTPUTSCHEMA_DESC = "Comma separated mapping of column names in the output schema to " +
+                                                   "the data types; for example: 'A:string,B:int'";
+    @Name(Constants.Reference.REFERENCE_NAME)
+    @Description(Constants.Reference.REFERENCE_NAME_DESCRIPTION)
+    public String referenceName;
+
+    @Name(IMPORT_QUERY)
+    @Description(IMPORT_QUERY_DESC)
+    @Nullable
+    @Macro
+    String importQuery;
+
+    @Name(PROJECT_ID)
+    @Description(PROJECTID_DESC)
+    @Macro
+    String projectId;
+
+    @Name(OUTPUT_SCHEMA)
+    @Description(OUTPUTSCHEMA_DESC)
+    @Macro
+    private String outputSchema;
+
+    @Name(INPUT_TABLE_ID)
+    @Description(INPUT_TABLE_DESC)
+    @Macro
+    String fullyQualifiedInputTableId;
+
+    @Name(JSON_FILE_PATH)
+    @Description(JSON_KEYFILE_DESC)
+    @Macro
+    String jsonFilePath;
+
+    @Name(TEMP_BUCKET)
+    @Description(TEMP_BUCKET_DESC)
+    @Macro
+    String tmpBucketPath;
+  }
+
+  private void getOutputSchema() {
+    if (outputSchema == null) {
+      List<Schema.Field> outputFields = Lists.newArrayList();
+      try {
+        for (String fieldName : outputSchemaMapping.keySet()) {
+          Schema fieldType = Schema.of(Schema.Type.valueOf(outputSchemaMapping.get(fieldName).toUpperCase()));
+          outputFields.add(Schema.Field.of(fieldName, fieldType));
+        }
+
+      } catch (Exception e) {
+        throw new IllegalArgumentException("Exception while creating output schema " +
+                                             "Invalid output " + "schema: " + e.getMessage(), e);
+      }
+      outputSchema = Schema.recordOf("outputSchema", outputFields);
+    }
+  }
+
```

Comment:
remove this line and add 1 after line 222, a class always ends with new line. checkstyle should have caught this